driver/bpf/plumbing_helpers.h

```diff
@@ -412,6 +464,204 @@ static __always_inline int init_filler_data(void *ctx,
+	case SYS_SENDMSG:
+		return PPME_SOCKET_SENDMSG_E;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
+	case SYS_SENDMMSG:
+		return PPME_SOCKET_SENDMMSG_E;
+#endif
+	case SYS_RECVMSG:
+		return PPME_SOCKET_RECVMSG_E;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+	case SYS_RECVMMSG:
+		return PPME_SOCKET_RECVMMSG_E;
+#endif
+	case SYS_ACCEPT4:
+		return PPME_SOCKET_ACCEPT4_5_E;
+	default:
+		return PPME_GENERIC_E;
+	}
+
+	return PPME_GENERIC_E;
+}
+
+/* Filter socketcalls based on the collector custom probe */
+static __always_inline enum ppm_event_type parse_socketcall_collector(int socketcall_id)
+{
+	switch (socketcall_id) {
+	case SYS_SOCKET:
+		return PPME_SOCKET_SOCKET_E;
+	case SYS_CONNECT:
+		return PPME_SOCKET_CONNECT_E;
+	case SYS_ACCEPT:
+		return PPME_SOCKET_ACCEPT_5_E;
+	case SYS_SHUTDOWN:
+		return PPME_SOCKET_SHUTDOWN_E;
+	case SYS_ACCEPT4:
+		return PPME_SOCKET_ACCEPT4_5_E;
+	default:
+		return PPME_GENERIC_E;
+	}
+
+	return PPME_GENERIC_E;
+}
+
+static __always_inline int __bpf_read_socketcall_args(void *dest, void *src, int sc_id)
+{
+	/* BPF verifier:
+	 *
+	 * Using nas[socketcall_id] causes an exception due to arithmetic operations
+	 * on the size argument. Therefore, use a switch statement instead, and directly
+	 * specify the number of bytes (arguments) to read.
+	 */
+#define AL(x) ((x) * sizeof(unsigned long))
+	switch (sc_id) {
+	case 0:
+		return 0;
+	case SYS_SOCKET:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_BIND:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_CONNECT:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_LISTEN:
+		return bpf_probe_read_user(dest, AL(2), src);
+	case SYS_ACCEPT:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_GETSOCKNAME:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_GETPEERNAME:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_SOCKETPAIR:
+		return bpf_probe_read_user(dest, AL(4), src);
+	case SYS_SEND:
+		return bpf_probe_read_user(dest, AL(4), src);
+	case SYS_RECV:
+		return bpf_probe_read_user(dest, AL(4), src);
+	case SYS_SENDTO:
+		return bpf_probe_read_user(dest, AL(6), src);
+	case SYS_RECVFROM:
+		return bpf_probe_read_user(dest, AL(6), src);
+	case SYS_SHUTDOWN:
+		return bpf_probe_read_user(dest, AL(2), src);
+	case SYS_SETSOCKOPT:
+		return bpf_probe_read_user(dest, AL(5), src);
+	case SYS_GETSOCKOPT:
+		return bpf_probe_read_user(dest, AL(5), src);
+	case SYS_SENDMSG:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_RECVMSG:
+		return bpf_probe_read_user(dest, AL(3), src);
+	case SYS_ACCEPT4:
+		return bpf_probe_read_user(dest, AL(4), src);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+	case SYS_RECVMMSG:
+		return bpf_probe_read_user(dest, AL(5), src);
+#endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
+	case SYS_SENDMMSG:
+		return bpf_probe_read_user(dest, AL(4), src);
+#endif
+	default:
+		return -1;
+	};
+#undef AL
+
+	return 0;
```

Comment:
[nit] I don't think we can get to this line, since all cases in the above switch-case return, including the `default` case. Is this here just to make the verifier happy or can we remove it?